File: katas/6 kyu/convert-string-to-camel-case/solution.py
import codewars_test as test
# -------------------------------- MY SOLUTION ------------------------------- #
def to_camel_case(text):
    removed = text.replace('-', ' ').replace('_', ' ').split()
    if len(removed) == 0:
        return ''
    return removed[0]+ ''.join([x.capitalize() for x in removed[1:]])
# ------------------------------ CLEVER SOLUTION ----------------------------- #
def to_camel_case(text):
    output = ''.join(x for x in text.title() if x not in "_-")
    return text[0] + output[1:] if text else ''
# ----------------------------------- TEST ----------------------------------- #
test.describe("Testing function to_camel_case")
test.it("Basic tests")
test.assert_equals(to_camel_case(''), '', "An empty string was provided but not returned")
test.assert_equals(to_camel_case("the_stealth_warrior"), "theStealthWarrior","to_camel_case('the_stealth_warrior') did not return correct value")
test.assert_equals(to_camel_case("The-Stealth-Warrior"), "TheStealthWarrior","to_camel_case('The-Stealth-Warrior') did not return correct value")
test.assert_equals(to_camel_case("A-B-C"), "ABC","to_camel_case('A-B-C') did not return correct value")

File: katas/6 kyu/convert-string-to-camel-case/solution.js
// convert-string-to-camel-case
/* ------------------------------- MY SOLUTION ------------------------------ */
function toCamelCase(str){
  arr = str.split("")
  result = []
  var i = 0
  while (i < arr.length) {
    var element = arr[i];
    if (element === "_" || element === "-") {
      element = arr[i+1].toUpperCase()
      i++
    }
    result.push(element)
    i++
  }
  return result.join("")
}
/* ----------------------------- CLEVER SOLUTION ---------------------------- */
function toCamelCase(str){
  var regExp=/[-_]\w/ig;
  return str.replace(regExp,function(match){
    return match.charAt(1).toUpperCase();
  });
}
/* ---------------------------------- TEST ---------------------------------- */
// const chai = require("chai");
// const Test = chai.assert.strictEqual; // replace Test.assertEqual with Test
// chai.config.truncateThreshold=0;
const { assert } = require('chai');

describe("Tests", () => {
  it("test", () => {
    assert.strictEqual(toCamelCase(''), '', "An empty string was provided but not returned")
    assert.strictEqual(toCamelCase("the_stealth_warrior"), "theStealthWarrior", "toCamelCase('the_stealth_warrior') did not return correct value")
    assert.strictEqual(toCamelCase("The-Stealth-Warrior"), "TheStealthWarrior", "toCamelCase('The-Stealth-Warrior') did not return correct value")
    assert.strictEqual(toCamelCase("A-B-C"), "ABC", "toCamelCase('A-B-C') did not return correct value")
  });
});
